src/module_2/module_2_EDA.md

```diff
@@ -0,0 +1,2197 @@
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>user_id</th>
+      <th>user_nuts1</th>
+      <th>first_ordered_at</th>
+      <th>customer_cohort_month</th>
+      <th>count_people</th>
+      <th>count_adults</th>
+      <th>count_children</th>
+      <th>count_babies</th>
+      <th>count_pets</th>
+    </tr>
+    <tr>
+      <th>user_segment</th>
+      <th></th>
+      <th></th>
+      <th></th>
+      <th></th>
+      <th></th>
+      <th></th>
+      <th></th>
+      <th></th>
+      <th></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>Proposition</th>
+      <td>2340</td>
+      <td>2315</td>
+      <td>2340</td>
+      <td>2340</td>
+      <td>193</td>
+      <td>193</td>
+      <td>193</td>
+      <td>193</td>
+      <td>193</td>
+    </tr>
+    <tr>
+      <th>Top Up</th>
+      <td>2643</td>
+      <td>2617</td>
+      <td>2643</td>
+      <td>2643</td>
+      <td>132</td>
+      <td>132</td>
+      <td>132</td>
+      <td>132</td>
+      <td>132</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+Más del 93% de los datos en 5 categorias son Null, parece que se dejaron de pedir por algun motivo (quizas temas de privacidad) o que se han pedido de manera opcional y pocos usuarios los proporcionan.
+
+# First Analysis conducted - Popularidad de producto
+En este análsis vamos a ver que productos son los más populares en función a las veces que estos son añadidos como 'wise to buy regularly'.
+
+
+```python
+df_variant_user_counts = df_regulars.groupby('variant_id')['user_id'].nunique().reset_index()
+df_variant_user_counts.rename(columns={'user_id': 'user_id_count'}, inplace=True)
+df_inventory_merged = pd.merge(df_inventory, df_variant_user_counts, how='left', on='variant_id')
+```
+
+
+```python
+print(df_inventory_merged.isnull().sum())
+```
+
+    variant_id            0
+    price                 0
+    compare_at_price      0
+    vendor                0
+    product_type          0
+    tags                  0
+    user_id_count       448
+    dtype: int64
+
+
+Parece que hay una serie de varianted el dataframe de inventory las cuales no estaban en el dataframe de regulars. Vamos a comprobarlo rapidamente para ver que estamos haciendo todo bien:
+
+
+```python
+print(f'Valores únicos de inventory: {len(df_inventory['variant_id'].unique())}')
+print(f'Valores únicos de regulars: {len(df_regulars['variant_id'].unique())}')
+```
+
+    Valores únicos de inventory: 1733
+    Valores únicos de regulars: 1843
+
+
+
+```python
+unicos_inventory = df_inventory['variant_id'].unique()
+unicos_regulars = df_regulars['variant_id'].unique()
+valores_coincidentes = np.intersect1d(unicos_inventory, unicos_regulars)
+
+print(f'Valores únicos coincidentes entre inventory y regulars: {len(unicos_inventory) - len(valores_coincidentes)}')
```

Comment:
The text doesn't reflect what you are computing here, you are computing: "Número de variantes que están en inventory pero no están en regulars`, so the "coincidentes" is wrong.

Reply:
I would probably compute this in the following way: `df_inventory[~ df_inventory["variant_id"].isin(df_regulars['variant_id'])].variant_id.nunique()`

Reply:
The most likely hypothesis about why there are some products in the inventory which doesn't have a counter part in the regulars is simply that these products are not popular enough to be added as regular by any user. This means that you should impute the NULL `user_id_count` as 0